### DISCUSSION TITLE
How to check v-model update?

### DISCUSSION THREAD

**Discussion by T0miii** [cypress-io/cypress#28454] [Component Testing]
I have a rather simpe Vue 3 Select Component 

![image](https://github.com/cypress-io/cypress/assets/10532141/a815d74a-0756-4a2c-a63f-38df7900f11c)

I click the input , dropdown opens, i click label value gets emited trough modelValue (v-model), new option is shown as selected.

How can i test this with cypress component tests

```
  it('renders', () => {
    let data = '3'
    cy.mount(MySelect, {
      props: {
        options,
        modelValue: data,
      },
    })

    cy.get('div.my-select').click()

    cy.get('div.my-dropdown__menu').should('be.visible')
    cy.get('li.my-select-dropdown__item').first().click()
  })
})
```

it mounts with the value 3 selected, but when the test opens the dropdown and click on another option the selected does not get updated.

**Comment by thomiNG** [comment]
I also ran into this problem. Does anyone know how update v-model?

**Comment by T0miii** [comment]
JSX was the only way we could make it work, Should be probably better described in the docs tho.

````jsx
import CheckboxComponent from './CheckboxComponent.vue'
import {ref} from "vue";

describe('<CheckboxComponent />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-vue
    const model = ref<unknown[]>([])
    cy.mount(() => (
        <CheckboxComponent v-model={model.value} label="Check This Item"/>
    ))
  })
})
````

**Reply by gigabites19** [reply]
Using `v-model` directly didn't work for me but this did:

```
// ...
  it('renders', () => {
    // see: https://on.cypress.io/mounting-vue
    const model = ref<unknown[]>([])
    cy.mount(() => (
        <CheckboxComponent
            modelValue={model.value}
            onUpdate:modelValue={(newValue) => model.value=newValue}
            label="Check This Item"
        />
    ))
  })
// ...
```

I had to also do:
`import { ref } from 'vue'` for this to run even though I am not using it directly. I am on Vue3/Vite and I enabled JSX only for cypress.
